pkg/autodiscovery/listeners/common.go

```diff
@@ -3,6 +3,8 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2017-present Datadog, Inc.
 
+// +build !serverless
```

Comment:
The clean way would probably be to move out the code using `k8s.io` behind the `kubeapiserver` and `kubelet` build flags, which are probably already out of the serverless build?

Reply:
Unfortunately kubelet is for example not excluded for the serverless build for instance here : https://github.com/DataDog/datadog-agent/blob/main/pkg/dogstatsd/enrich.go
But more PR are coming to have a better code split